pkg/util/validation/limits.go

```diff
@@ -185,6 +185,7 @@ type Limits struct {
 	RulerRecordingRulesEvaluationEnabled bool           `yaml:"ruler_recording_rules_evaluation_enabled" json:"ruler_recording_rules_evaluation_enabled" category:"experimental"`
 	RulerAlertingRulesEvaluationEnabled  bool           `yaml:"ruler_alerting_rules_evaluation_enabled" json:"ruler_alerting_rules_evaluation_enabled" category:"experimental"`
 	RulerSyncRulesOnChangesEnabled       bool           `yaml:"ruler_sync_rules_on_changes_enabled" json:"ruler_sync_rules_on_changes_enabled" category:"advanced"`
+	RulerMaxRulesPerRuleGroupByNamespace LimitsMap[int] `yaml:"ruler_max_rules_per_rule_group_by_namespace" json:"ruler_max_rules_per_rule_group_by_namespace" category:"advanced"`
```

Comment:
I would mark it experimental first and list it among experimental features in `docs/.../about-versioning.md`. Once the entire work is done and we're happy with it, then we can switch it to advanced.

Reply:
Done in https://github.com/grafana/mimir/pull/8378/commits/7e45e06d204213f3e7125ca5fd72022e43c50e08

When CI passes, I'll merge this.